File: basics/Q41/main.go
package main

import (
	"fmt"
	"os"
	"path/filepath"
)

func main() {
	exePath, err := os.Executable()
	if err != nil {
		fmt.Println("Error:", err)
		return
	}
	absPath, err := filepath.Abs(exePath)
	if err != nil {
		fmt.Println("Error:", err)
		return
	}

	fmt.Println("Current File Name", absPath)
}

File: basics/Q41/main.rs
use std::env;

fn main() {
    let exe_path = match env::current_exe() {
        Ok(p) => p,
        Err(e) => {
            println!("Error: {}", e);
            return;
        }
    };

    let abs_path = match exe_path.canonicalize() {
        Ok(p) => p,
        Err(e) => {
            println!("Error: {}", e);
            return;
        }
    };

    println!("Current File Name {}", abs_path.display());
}